### DISCUSSION TITLE
Issu with OpenAI key

### DISCUSSION THREAD

**Discussion by divya-akula** [microsoft/hack-together-teams#62] [Q&A] [answered]
hi @waldekmastykarz / @aycabas  ,

  I am using the openAI key which I got from the portal Azure it gives me the bot error

![image](https://github.com/microsoft/hack-together-teams/assets/9211327/4732c283-1a8f-4691-9b34-211e38b2c671)

![image](https://github.com/microsoft/hack-together-teams/assets/9211327/5450ef49-df96-4dab-b526-6e0872524b3c)

**Comment by aycabas** [comment]
Hi @divya-akula can you please share how you consumed the key in your code? Are you using Teams AI library with Azure Open AI key?

**Comment by divya-akula** [comment]
Yeah I am

**Comment by divya-akula** [comment]
![image](https://github.com/microsoft/hack-together-teams/assets/9211327/eb3d2968-c972-4ea4-92f2-c84a873621e9)


![image](https://github.com/microsoft/hack-together-teams/assets/9211327/9982569f-6cbf-4cd7-acaa-c9495e9fe85d)

**Comment by divya-akula** [comment]
Hi @aycabas/ @waldekmastykarz ,

  The issue has been fixed , thanks

**Reply by waldekmastykarz** [reply]
Cool! Would you be willing to share how you solved it in case someone else comes across the same issue?

**Reply by divya-akula** [reply, marked as answer]
hi @waldekmastykarz ,

  Sure, I would love to share,I was using Azure OpenAI key with OpenAI , and the above code would remain the same, except that the planner am using now is Azure Planner
``` typescript
const planner = new AzureOpenAIPlanner({
  apiKey: config.openAIKey,
  defaultModel: 'MyGPT3', // Model created in GPT
  logRequests: true,
  endpoint:config.openAPIEndPoint,
});
```
and updated the references and installed the modules accordingly

``` typescript
import { AzureOpenAIClient, AzureOpenAIClientOptions } from "@microsoft/teams-ai/lib/OpenAIClients";
import { AzureKeyCredential, OpenAIClient } from "@azure/openai";
import { Application, ConversationHisory, DefaultPromptManager, DefaultTurnState, OpenAIModerator,AzureOpenAIPlannerOptions, AzureOpenAIPlanner, AI } from '@microsoft/teams-ai';

```

**Reply by divya-akula** [reply]
And also removed moderation , doesnt seem it works with Azure OpenAI , yet
